internal/pkg/agent/vault/vault_linux.go

```diff
@@ -157,3 +224,40 @@ func deriveKey(pw []byte, salt []byte) ([]byte, []byte, error) {
 func (v *Vault) filepathFromKey(key string) string {
 	return filepath.Join(v.path, fileNameFromKey(v.key, key))
 }
+
+// writeFile "atomic" file write, utilizes temp file and replace
+// os.CreateTemp creates the file with 0600 mask, which is what we need
+func writeFile(fp string, data []byte) (err error) {
+	dir, fn := filepath.Split(fp)
+	if dir == "" {
+		dir = "."
+	}
+
+	f, err := os.CreateTemp(dir, fn)
+	if err != nil {
+		return fmt.Errorf("failed creating temp file: %w", err)
+	}
+	defer func() {
+		if err != nil {
+			_ = os.Remove(f.Name())
```

Comment:
[Blocker]
Please add this to the error as well.
You can join this error with the other and an extra message to clearly states this error is from the cleanup.

```suggestion
			rerr := os.Remove(f.Name())
			if rerr !=nil{
				err= errors.Join(err, fmt.Errorf("cleanup failed, could not remove temp file: %w", rerr))
			}
```